apps/condo/domains/notification/adapters/redStoreAdapter.js

```diff
@@ -0,0 +1,157 @@
+ * @param data
+ */
+function prepareData (data = {}, token) {
+    const result = { token }
+
+    for (const key in data) {
+        if (data.hasOwnProperty(key)) result[key] = String(data[key])
+    }
+
+    return result
+}
+/**
+ * Send push notification to pushToken via app, configured by REDSTORE_CONFIG in .helm (.env)
+ * Attention! Notifications could only be sent to devices, connected via same PROJECT_ID.
+ * Attempts to send push notifications to devices, connected through different projects will fail.
+ */
+class RedStoreAdapter {
+    _config = null
+
+    constructor (config = REDSTORE_CONFIG) {
+        try {
+            if (isEmpty(config)) throw new Error(`Valid ${REDSTORE_CONFIG_ENV} should be provided within .helm (.env)`)
+        } catch (err) {
+            // For CI/local tests config is useless because of emulation via FAKE tokens
+            logger.error({ msg: 'redStore adapter error', err })
+        }
+
+        this._config = config
+    }
+
+    /**
+     * Validates and prepares notification significant fields
+     * @param title
+     * @param body
+     * @returns {{title, body}}
+     */
+    static validateAndPrepareNotification ({ title, body } = {}) {
+        if (!title || !body || isEmpty(title) || isEmpty(body)) throw new Error('Missing notification.title or notification.body')
+
+        return { title, body }
+    }
+
+    /**
+     * Prepares notification for either/both sending to redStore
+     * Converts single notification to notifications array (for multiple tokens provided) for batch request
+     * @param notificationRaw
+     * @param data
+     * @param tokens
+     * @param pushTypes
+     * @param isVoIP
+     * @returns {*[][]}
+     */
+    static prepareBatchData (notificationRaw, data, tokens = [], pushTypes = {}, appIds = {}, isVoIP = false) {
+        const notification = RedStoreAdapter.validateAndPrepareNotification(notificationRaw)
+        const notifications = [] // User can have many Remote Clients. Message is created for the user, so from 1 message there can be many notifications
+        const pushContext = {}
+
+        tokens.forEach((pushToken) => {
+            const pushType = pushTypes[pushToken] || PUSH_TYPE_DEFAULT
+            const preparedData = prepareData(data, pushToken)
+            const pushData = {
+                token: pushToken,
+                data: preparedData,
+                notification,
+                appId: appIds[pushToken],
+                ...DEFAULT_PUSH_SETTINGS,
+            }
+
+            if (!APPS_WITH_DISABLED_NOTIFICATIONS.includes(pushData.appId)) notifications.push(pushData)
+
+            if (!pushContext[pushType]) pushContext[pushType] = pushData
+        })
+
+        return [notifications, pushContext]
+    }
+
+    /**
+     * Manages to send notification to all available pushTokens of the user.
+     * @param notification
```

Comment:
JSDoc require to describe parameters, otherwise why you need such documentation without any description?